pkg/profilecreator/profilecreator.go

```diff
@@ -266,31 +295,38 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 		} else {
 			max = reservedCPUCount
 		}
-		//sorting by comparing the first logical core of processor cores
-		sort.Slice(node.Cores, func(i, j int) bool {
-			//sorting the logical cores a processor
-			sort.Slice(node.Cores[i].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[i].LogicalProcessors[x] < node.Cores[i].LogicalProcessors[y]
-			})
-			sort.Slice(node.Cores[j].LogicalProcessors, func(x, y int) bool {
-				return node.Cores[j].LogicalProcessors[x] < node.Cores[j].LogicalProcessors[y]
-			})
-			return node.Cores[i].LogicalProcessors[0] < node.Cores[j].LogicalProcessors[0]
-		})
+		if max%2 != 0 && htEnabled {
+			return "", "", fmt.Errorf("Can't allocatable odd number of CPUs from a NUMA Node")
+		}
 		for _, cores := range node.Cores {
 			for _, core := range cores.LogicalProcessors {
+				totalCPUSet.Add(core)
 				if reservedCPUSet.Result().Size() < max {
 					reservedCPUSet.Add(core)
-				} else {
-					isolatedCPUSet.Add(core)
 				}
 			}
 		}
 	}
+	isolatedCPUSet := totalCPUSet.Result().Difference(reservedCPUSet.Result())
+	log.Infof("reservedCPUs: %v len(reservedCPUs): %d\n isolatedCPUs: %v len(isolatedCPUs): %d\n", reservedCPUSet.Result().String(), reservedCPUSet.Result().Size(), isolatedCPUSet.String(), isolatedCPUSet.Size())
+	return reservedCPUSet.Result().String(), isolatedCPUSet.String(), nil
 
-	log.Infof("reservedCPUs: %v len(reservedCPUs): %d \n", reservedCPUSet.Result().String(), reservedCPUSet.Result().Size())
-	log.Infof("isolatedCPUs: %v len(isolatedCPUs): %d \n", isolatedCPUSet.Result().String(), isolatedCPUSet.Result().Size())
+}
+func (ghwHandler GHWHandler) isHyperthreadingEnabled() (bool, error) {
+	cpuInfo, err := ghwHandler.CPU()
+	if err != nil {
+		return false, fmt.Errorf("Error obtaining CPU Info from GHW snapshot: %v", err)
+	}
+	return contains(cpuInfo.Processors[0].Capabilities, "ht"), nil
```

Comment:
this is ok. It however deserves a comment about why we pick the 0th processor, something like

"since there is no way to disable flags per-processor (not system wide) we check the flags of the first available processor"

Reply:
A following implementation will leverage the /sys/devices/system/cpu/smt/active file which is the "standard" way to query HT.

Reply:
however I was doublechecking this and we need to be at least aware of the fine details between firmware settings (aka HT enabled/disabled in BIOS or UEFI) and kernel settings.
1. the HT flag, if present, is telling us that the HW is *ht capable*
2. the "nosmt" kernel parameters seems (didn't dive deep enough yet) to do something very close to just offline the hyperthread sibilings.

So in practice:
1. if the BIOS and the kernel aligns, checking for the presence of HT flags tells us if the system is running with SMT enabled - or not
2. if the BIOS and the kernel DO NOT align, things get interesting. The only meaningful case is: HT enabled in firmware, `nosmt` given. In this case, this function wrongly reports the HT state. For userspace, HT is enabled only if ALL hardware, firmware and kernel allow + enable SMT. So in this case the computation will be wrong.

But this is actually no different from the corner case on which everything is enabled, yet some cpu cores are offlined for whatever reason.

So reading `/sys/devices/system/cpu/smt/active` seems to be the single action with more ROI, but this will require some changes in how PPC handles the snapshots.

Reply:
nosmt kernel arg does the same as editing `echo [on|off] > /sys/devices/system/cpu/smt/control`, enables a single thread per core.
BTW, the option we have in PPC is "disable-ht=true/false" and not "ht=true/false" because of the reasons you mentioned.
We will need to add this file to the ghw snapshot.

Reply:
done: https://github.com/openshift-kni/performance-addon-operators/pull/567
I'm still not sure reading `/sys/devices/system/cpu/smt/active` alone is everything we need, but I concur that is probably the single best option.

Reply:
Sure, will add a comment.

Reply:
@fromanirh so fast! thanks!